python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -3423,8 +3424,6 @@ def test_all(data):
             expected = pdata.all(bool_only=True)
             assert_eq(got, expected)
         else:
-            with pytest.raises(NotImplementedError):
-                gdata.all(bool_only=False)
```

Comment:
Is this a behaviour change? (ditto for code removed in tests below)

Reply:
Ah yes I suppose that it is, it's disabling a failure that was previously present but didn't make sense. `bool_only` is not implemented for `Series` in `pandas` because the purpose is to filter a `DataFrame` to only perform the reduction on boolean column types. However, because in the previous version of the code `DataFrame.any` would call through to `Series.any` and forward the argument through, it would fail, despite the fact that `DataFrame.any` contained the logic to do the filtering to boolean columns. Now that is fixed.

The code removed in tests below is from when I originally removed those Index reductions. I'll add them back now that we've decided to reenable those reductions.